src/classes/i18n.js

```diff
@@ -199,15 +209,94 @@ export default class i18n {
     }
 
     get_lang_files (lang, cb) {
+        const MAX_RETRY = this.config.get('download_retry');
 
         // append a random string, to avoid getting a response from cache
         const random_str = '&rand=' + ~~(Math.random() * 1000);
         const url = this.translations_url.replace(':lang', lang) + random_str;
-        const write_stream = fs.createWriteStream(this.config.get('locale_dir') + lang + '.json');
 
-        request(url)
-            .pipe(write_stream)
-            .on('finish', cb);
+        const translation_file_path = this.config.get('locale_dir') + lang + '.json';
+
+        new Promise(async (resolve, reject) => {
+            if (
+                process.env.REFRESH_TRANSLATIONS
+                || !await this.version_match(translation_file_path)
+            ) {
+                // force re-download translations
+                await this.download_translations(url, translation_file_path);
+            }
+
+
+            let retry_count = 0;
+
+            async.doWhilst(
```

Comment:
To consider

Where else do we use async? Perhaps upgrade all lib to latest while we are at it?

Reply:
Hmm, not sure which libs where else we're using async. I'll create a ticket for this.

Reply:
For now a placeholder. https://freedom.myjetbrains.com/youtrack/issue/Dev-3939 while we identify our libs using async that needs to be update.

Reply:
I meant inside i18n..we don't need to update other lib?

Reply:
oh.. okay.. that "all lib" was confusing. Alright.. I'll take a look at upgrading the libs in this module.

Reply:
Sorry for the confusion

Reply:
Update modules to use latest version https://github.com/anyTV/anytv-i18n/pull/10/commits/276fe04fc5d6ae2314944e85ad963f264c9493a3